src/codons.py

```diff
@@ -35,8 +35,19 @@ def split_codons(dna: str) -> list[str] | None:
     True
 
     """
-    # FIXME: Implement the function
-    return []
+
+    if dna == "":
+        return []
+
+    if (len(dna) % 3) != 0:
+        return None
+
+    triplets = []
+
+    for i in range(0,len(dna),3):
+        triplets.append(dna[i : i+3])
+
+    return triplets
```

Comment:
This looks perfect as it is, but if you want it a little shorter, you can use a list comprehension

```python
return [ dna[i : i+3] for i in range(0, len(dna), 3) ]
```